pkg/intercp/tls/defaults.go

```diff
@@ -0,0 +1,62 @@
+package tls
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/sethvargo/go-retry"
+
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/resources/store"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/core/user"
+)
+
+type DefaultsComponent struct {
```

Comment:
All this code is almost identical to EnvoyAdminCaDefaultComponent and friends. Is there no possible way to share/reuse some of the existing code?

Reply:
friends?

I can try to generify `tls` package (pki+defaults) between the Envoy Admin and this.
Do you want me to do this in this PR?

Reply:
Up to you just a suggestion. Just feels this code is very similar

Reply:
ok, then skipping this change now. Will revisit it after the task is finished.